test/unit/android-helper-specs.js

```diff
@@ -101,82 +101,9 @@ describe('Android Helpers', () => {
-      mocks.adb.expects('reboot').never();
-      await helpers.ensureDeviceLocale(adb, 'en', 'us');
-      mocks.adb.verify();
-    });
-    it('should set language and country if they are different when API < 23', async () => {
-      mocks.adb.expects('getApiLevel').returns(18);
-      mocks.adb.expects('getDeviceLanguage').returns('fr');
-      mocks.adb.expects('getDeviceCountry').returns('FR');
-      mocks.adb.expects('getDeviceLocale').never();
-      mocks.adb.expects('setDeviceLanguage').withExactArgs('en')
-        .returns("");
-      mocks.adb.expects('setDeviceCountry').withExactArgs('us')
-        .returns("");
-      mocks.adb.expects('setDeviceLocale').never();
-      mocks.adb.expects('reboot').returns(null);
-      await helpers.ensureDeviceLocale(adb, 'en', 'us');
-      mocks.adb.verify();
-    });
-    it('should not set locale if it does not change when API = 23', async () => {
-      mocks.adb.expects('getApiLevel').returns(23);
-      mocks.adb.expects('getDeviceLanguage').never();
-      mocks.adb.expects('getDeviceCountry').never();
-      mocks.adb.expects('getDeviceLocale').returns('en-US');
-      mocks.adb.expects('setDeviceLanguage').never();
-      mocks.adb.expects('setDeviceCountry').never();
-      mocks.adb.expects('setDeviceLocale').never();
-      mocks.adb.expects('reboot').never();
-      await helpers.ensureDeviceLocale(adb, 'en', 'us');
-      mocks.adb.verify();
-    });
-    it('should set locale if it is different when API = 23', async () => {
-      mocks.adb.expects('getApiLevel').returns(23);
-      mocks.adb.expects('getDeviceLanguage').never();
-      mocks.adb.expects('getDeviceCountry').never();
-      mocks.adb.expects('getDeviceLocale').returns('fr-FR');
-      mocks.adb.expects('setDeviceLanguage').never();
-      mocks.adb.expects('setDeviceCountry').never();
-      mocks.adb.expects('setDeviceLocale').withExactArgs('en-US')
-        .returns("");
-      mocks.adb.expects('reboot').returns(null);
-      await helpers.ensureDeviceLocale(adb, 'en', 'us');
-      mocks.adb.verify();
-    });
-    it('should set locale to country only if language is not passed when API = 23', async () => {
-      mocks.adb.expects('getApiLevel').returns(23);
-      mocks.adb.expects('getDeviceLocale').returns('fr-FR');
-      mocks.adb.expects('setDeviceLocale').withExactArgs('en').returns("");
-      mocks.adb.expects('reboot').returns(null);
-      await helpers.ensureDeviceLocale(adb, 'en');
-      mocks.adb.verify();
-    });
-    it('should set locale to language only if country is not passed when API = 23', async () => {
-      mocks.adb.expects('getApiLevel').returns(23);
-      mocks.adb.expects('getDeviceLocale').returns('fr-FR');
-      mocks.adb.expects('setDeviceLocale').withExactArgs('US').returns("");
-      mocks.adb.expects('reboot').returns(null);
-      await helpers.ensureDeviceLocale(adb, null, 'US');
+    it('should call setDeviceLanguageCountry', async () => {
```

Comment:
Can you also add at least one negative test, where given values are not acceptable?

Reply:
Also, do we need to do any changes to UIA2 driver in order to update locale setting or it reuses the helper method from this module

Reply:
I found `ensureDeviceLocale` in UIA2 driver's test code and skip it for API > 23.
I'll create a PR for updating `package.json` and fixing the test code.

https://github.com/appium/appium-uiautomator2-driver/blob/d62a29d8fcf216acb5a663ab04da0cade516be57/test/functional/commands/language-e2e-specs.js#L27